.github/workflows/apidiff.yml

```diff
@@ -19,7 +19,7 @@ jobs:
       - name: Setup Go
         uses: actions/setup-go@v2
         with:
-          go-version: '1.17'
+          go-version: '1.18'
       - name: Execute go-apidiff
         uses: joelanford/go-apidiff@v0.2.0
         with:
```

Comment:
Question: should we define it as a breaking change?
If yes, why? What breaks for those who scaffold the project with the previous versions and now will use the next KB CLI version with this bump? Should not be it an `:sparkle:`?

Reply:
TBH, I wasn't sure also how to mark it:

- It's marked as breaking changes because it requires a go 1.18 for new projects. Also, there's a high possibility we'll want to use Go 1.18 features such as generics.
- Other than that, old projects will be able to continue to use it.

WDYT?

Reply:
I do not think that it is a breaking change ( indeed we cannot introduce breaking changes )
Why?
Breaking change will be only considered when we do a change that breaks something for those who scaffold a project with a KB version previously and then by getting the latest KB with the change will face some issue that is not the case. See: https://github.com/kubernetes-sigs/kubebuilder/blob/master/VERSIONING.md

Also, we have no generics in the scaffolds. Ditto so, when/if we add any implementation that uses it by default would be a breaking change and could only be introduced in a new /future golang plugin version such as go/v4.

Reply:
ok.  I changed the title.

I'm not sure why do you want to change the Github action go version...

Reply:
To ensure that we test the project with the go version used by the project